ext/drreg/drreg.c

```diff
@@ -2004,10 +3199,9 @@ drreg_exit(void)
 
     drmgr_exit();
 
-    if (ops.num_spill_slots > 0) {
-        if (!dr_raw_tls_cfree(tls_slot_offs, ops.num_spill_slots))
-            return DRREG_ERROR;
-    }
+    /* +1 for the pointer to the indirect spill block, see above. */
+    if (!dr_raw_tls_cfree(tls_simd_offs, ops.num_spill_slots + 1))
+        return DRREG_ERROR;
```

Comment:
Hmm given the huge increase in the size of the file: would it make sense to have a separate drreg_simd.c just to keep things more manageable and readable?  Just a thought.

Reply:
Yes, actually I had the same thought. This might be possible. Maybe also drreg_gpr.c and keep drreg.c as the main driver?

Reply:
Not sure if we should split further for flags... drreg_flags.c?

Reply:
SGTM.  More work: presumably a separate PR to split gpr and flags with some shared headers.  But might be nicer in the long run.  One downside is that previously static symbols are now visible, and in C there's nothing in between file-static and global, so we might have to be more careful with names.